presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/db/DbResourceGroupConfigurationManager.java

```diff
@@ -0,0 +1,296 @@
+import com.facebook.presto.execution.resourceGroups.ResourceGroupConfigurationManager;
+import com.facebook.presto.execution.resourceGroups.ResourceGroupId;
+import com.facebook.presto.execution.resourceGroups.ResourceGroupSelector;
+import com.facebook.presto.execution.resourceGroups.SelectionContext;
+import com.facebook.presto.execution.resourceGroups.StaticSelector;
+import com.facebook.presto.memory.ClusterMemoryPoolManager;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Sets;
+import io.airlift.units.DataSize;
+import io.airlift.units.Duration;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.annotation.concurrent.GuardedBy;
+import javax.inject.Inject;
+import javax.inject.Provider;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.PriorityQueue;
+import java.util.Set;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static com.facebook.presto.memory.LocalMemoryManager.GENERAL_POOL;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static io.airlift.concurrent.Threads.daemonThreadsNamed;
+import static io.airlift.units.DataSize.Unit.BYTE;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.Executors.newSingleThreadScheduledExecutor;
+
+public class DbResourceGroupConfigurationManager
+        implements ResourceGroupConfigurationManager, Provider<List<? extends ResourceGroupSelector>>
+{
+    private ResourceGroupsDao dao;
+    private final Map<ResourceGroupId, ConfigurableResourceGroup> groups = new HashMap<>();
+    @GuardedBy("generalPoolMemoryFraction")
+    private final Map<ConfigurableResourceGroup, Double> generalPoolMemoryFraction = new HashMap<>();
+    @GuardedBy("generalPoolMemoryFraction")
+    private long generalPoolBytes;
+
+    private List<ResourceGroupRecord> rootGroups;
+    private Map<String, ResourceGroupRecord> groupMap = new HashMap<>();
+    private Map<String, List<ResourceGroupId>> activeGroups = new HashMap<>();
+    private List<? extends ResourceGroupSelector> selectors;
+    private Optional<Duration> cpuQuotaPeriod;
+    private final ScheduledExecutorService configExecutor = newSingleThreadScheduledExecutor(daemonThreadsNamed("DbResourceGroupConfigurationManager"));
+    private final AtomicBoolean started = new AtomicBoolean();
+
+    @Inject
+    public DbResourceGroupConfigurationManager(ClusterMemoryPoolManager memoryPoolManager, Provider<ResourceGroupsDao> daoProvider)
+    {
+        requireNonNull(memoryPoolManager, "ClusterMemoryPoolManager is null");
```

Comment:
Usually we write these error messages to match the variable name, so `"memoryPoolManager is null"`